ext/standard/rand.c

```diff
@@ -372,6 +389,95 @@ PHP_FUNCTION(mt_getrandmax)
 }
 /* }}} */
 
+/* {{{ proto string rand_bytes(int length, int source)
+   Create a string of random bytes */
+PHP_FUNCTION(rand_bytes)
+{
+	char *bytes;
+	zend_long source = URANDOM;
+	zend_long length;
+	int n = 0;
+	int fd = -1;
+	size_t read_bytes = 0;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l|l", &length, &source) == FAILURE) {
+		return;
+	}
+
+	if (length <= 0 || length >= INT_MAX) {
+		php_error_docref(NULL, E_WARNING, "Cannot create a random string with a length of less than 1 or greater than %d", INT_MAX);
+		RETURN_FALSE;
+	}
+
+	bytes = ecalloc(length + 1, 1);
+
+	switch(source) {
+#if PHP_WIN32
+		case ARANDOM:
+		case RANDOM:
+		case URANDOM:
+			BYTE *bytes_b = (BYTE *) iv;
+			if (php_win32_get_rand_bytes(bytes_b, (size_t)length) == FAILURE){
+				efree(bytes);
+				php_error_docref(NULL, E_WARNING, "Could not gather sufficient random data");
+				RETURN_FALSE;
+			}
+			n = (int)length;
+			break;
+#else
+		case ARANDOM:
+			fd = open("/dev/arandom", O_RDONLY);
+		case RANDOM:
```

Comment:
Newer linux versions also have a syscall for getting random numbers (that iirc even has sane behavior), maybe we should try that before anything else on platforms that support it?